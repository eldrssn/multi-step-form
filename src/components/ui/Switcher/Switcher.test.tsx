import { fireEvent, render } from '@testing-library/react';
import { Switcher } from './';

describe('correct display Switcher when switching', () => {
  test('if position is off', () => {
    const { getByRole, container } = render(
      <Switcher isChecked={false} handleToggle={() => {}} />
    );

    expect(getByRole('checkbox')).not.toHaveAttribute('checked');
    expect(container).toMatchSnapshot();
  });

  test('if position is on', () => {
    const { getByRole } = render(
      <Switcher isChecked={true} handleToggle={() => {}} />
    );

    expect(getByRole('checkbox')).toHaveAttribute('checked');
  });

  test('if switcher is clicked', () => {
    const handleToggleMocked = jest.fn();

    const { getByRole } = render(
      <Switcher isChecked={true} handleToggle={handleToggleMocked} />
    );

    const switcher = getByRole('checkbox');
    fireEvent.click(switcher);
    expect(handleToggleMocked).toBeCalledTimes(1);
  });
});
