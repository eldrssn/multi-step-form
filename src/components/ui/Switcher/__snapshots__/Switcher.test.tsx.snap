// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct display Switcher when switching if position is off 1`] = `
<div>
  <label
    class="toggle_switch"
  >
    <input
      class="toggle_input"
      type="checkbox"
    />
    <span
      class="slider"
    />
  </label>
</div>
`;
